### PR TITLE
IDEMPIERE-5893 Completed invoices are being updated in MOrder.afterSave

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5893

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] I have checked my code and corrected any misspellings
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
### Tests
- [x] I have tested the direct scenario that my code is solving
- [x] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [x] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/MOrder.java [hunk 1]
```diff
@@ -1406,49 +1406,41 @@ protected boolean afterSave (boolean newRecord, boolean success)
...
-	      
+
+		// Propagate changes to not-completed/reversed/closed invoices
+		String[] propagateCols = new String[] {
+				COLUMNNAME_Description,	
+				COLUMNNAME_POReference,
+				COLUMNNAME_PaymentRule,
+				COLUMNNAME_C_PaymentTerm_ID,
+				COLUMNNAME_DateAcct};
+		boolean propagateColChanged = false;
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/model/MOrder.java:1417]: what would be the options available if an implementation wish to change the auto propagate columns ?

Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/compiere/model/MOrder.java:1417]: more columns can be propagated with an EventHandler

less columns or a different document status would not be possible, maybe an EventHandler on MInvoice to capture the change and reset it back to the old value?

any suggestion?

Comment (hengsin) [reply org.adempiere.base/src/org/compiere/model/MOrder.java:1417]: perhaps as a new sysconfig entry ?

Comment (CarlosRuiz-globalqss) [reply org.adempiere.base/src/org/compiere/model/MOrder.java]: > perhaps as a new sysconfig entry ?

ready, changed the approach to use SysConfig ORDER_COLUMNS_TO_COPY_TO_NOT_COMPLETED_INVOICES

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**